File: 3396. Minimum Number of Operations to Make Elements in Array Distinct/main.py
from collections import defaultdict
from math import ceil
from typing import List


class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        cnt = defaultdict(int)
        idx = -1
        for i in range(len(nums) - 1, -1, -1):
            cnt[nums[i]] += 1
            if cnt[nums[i]] == 2:
                idx = i
                break
        return ceil((idx + 1) / 3)

File: 3396. Minimum Number of Operations to Make Elements in Array Distinct/main.go
package main

import "math"

func minimumOperations(nums []int) int {
	cnt := make(map[int]int)
	idx := -1
	for i := len(nums) - 1; i > -1; i-- {
		cnt[nums[i]]++
		if cnt[nums[i]] == 2 {
			idx = i
			break
		}
	}
	return int(math.Ceil(float64((idx + 1)) / 3.0))
}
